src/configLoader.js

```diff
@@ -13,12 +14,20 @@ export function configLoader(configPath) {
         return process.exit(1);
     }
 
+    const rawConfig = fs.readFileSync(configPath, 'utf8');
+    const compiledConfig = rawConfig.replace(/\$\{([A-Za-z_]+)\}\$/g, (match, variableName) => {
+        if (process.env[variableName] === undefined) {
+            throw new Error(`Configuration value ${variableName} was not present in the environment`)
+        }
+        return process.env[variableName];
+    });
+
     if(/(\.yml)|(\.yaml)/.test(configPath)) {
-        return yaml.safeLoad(fs.readFileSync(configPath));
+        return yaml.safeLoad(compiledConfig);
     }
 
     if (/(\.json)/.test(configPath)) {
-        return JSON.parse(fs.readFileSync(configPath));
+        return JSON.parse(compiledConfig);
     }
 
     if (/(\.js)/.test(configPath)) {
```

Comment:
There's a good chance that generating `compiledConfig` will error for a `.js` config file, even though it's not being used. Can we either drop support for a js config file, or only generate the compiled config within the conditionals on L25/L29?

Reply:
oh yeah - thanks for bringing this up, i meant to ask about it. i would also like to drop support for `js` files if we can. honestly don't think we even need JSON? but it's a nice backup and no that different from YAML, in terms of logic

Reply:
🎉 I'm not too worried about JSON: YAML is a superset of it, so I don't think there's really any added complexity to supporting it as an input format.